src/posthog-surveys.ts

```diff
@@ -89,6 +92,20 @@ export class PostHogSurveys {
                     : true
                 return urlCheck && selectorCheck
             })
+
+            const eventBasedSurveys = conditionMatchedSurveys.filter((survey) => {
+                return survey.events && survey.events.length > 0
+            })
+
+            const eventsReceiver = new SurveyEventReceiver()
```

Comment:
`getActiveMatchingSurveys` is called every 3 seconds when using the popup. It doesn't seem right to initialise a new receiver every time we try to get active flags.

I think the way this should work is:

1. SurveyEventReceiver is class property, initialised once.
2. We find and register event based surveys in `getSurveys`, not in `getActiveMatchingSurveys` - because the survey list can only change when `getSurveys` is called, and is forced to fetch new surveys from the API.
3. In `getActiveMatchingSurveys` here, we only call `eventsReceiver.getSurveys()`.